modules/openxr/action_map/openxr_interaction_profile.cpp

```diff
@@ -65,38 +69,60 @@ Ref<OpenXRAction> OpenXRIPBinding::get_action() const {
 	return action;
 }
 
-int OpenXRIPBinding::get_path_count() const {
-	return paths.size();
+void OpenXRIPBinding::set_source_path(const String &path) {
+	source_path = path;
+	emit_changed();
 }
 
-void OpenXRIPBinding::set_paths(const PackedStringArray p_paths) {
-	paths = p_paths;
-	emit_changed();
+String OpenXRIPBinding::get_source_path() const {
+	return source_path;
 }
 
-PackedStringArray OpenXRIPBinding::get_paths() const {
+void OpenXRIPBinding::set_paths(const PackedStringArray p_paths) { // deprecated
+	// Fallback logic, this should ONLY be called when loading older action maps.
+	// We'll parse this momentarily and extract individual bindings.
+	source_path = "";
+	for (const String &path : p_paths) {
+		if (!source_path.is_empty()) {
+			source_path += ",";
+		}
+		source_path += path;
+	}
+}
+
+PackedStringArray OpenXRIPBinding::get_paths() const { // deprecated
+	// Fallback logic, return an array.
+	// If we just loaded an old action map from disc, this will be a comma separated list of actions.
+	// Once parsed there should be only one path in our array.
+	PackedStringArray paths = source_path.split(",", false);
+
 	return paths;
 }
 
-void OpenXRIPBinding::parse_paths(const String p_paths) {
-	paths = p_paths.split(",", false);
-	emit_changed();
+int OpenXRIPBinding::get_path_count() const { // deprecated
+	// Fallback logic, we only have one entry.
+	return source_path.is_empty() ? 0 : 1;
 }
 
-bool OpenXRIPBinding::has_path(const String p_path) const {
-	return paths.has(p_path);
+bool OpenXRIPBinding::has_path(const String p_path) const { // deprecated
+	// Fallback logic, return true if this is our path.
+	return source_path == p_path;
 }
 
-void OpenXRIPBinding::add_path(const String p_path) {
-	if (!paths.has(p_path)) {
-		paths.push_back(p_path);
+void OpenXRIPBinding::add_path(const String p_path) { // deprecated
+	// Fallback logic, only assign first time this is called.
+	if (source_path != p_path) {
+		ERR_FAIL_COND(!source_path.is_empty());
+
+		source_path = p_path;
 		emit_changed();
 	}
 }
 
-void OpenXRIPBinding::remove_path(const String p_path) {
-	if (paths.has(p_path)) {
-		paths.erase(p_path);
+void OpenXRIPBinding::remove_path(const String p_path) { // deprecated
+	// Fallback logic, clear if this is our path.
+	if (source_path == p_path) {
```

Comment:
An error message would be nice here for when `source_path != p_path` for the same reason, since existing code dealing with multiple paths would need to be rewritten.